### PR TITLE
Integrate new incoming headers

### PR DESCRIPTION
**Pull request by mariusmihaic**
## Reasoning behind the pull request
- Receive incoming headers notifications from sovereign notifier. 
  
## Proposed changes
- Created an `incomingHeaderHandler` comp which will be notified by the sovereign notifier events w.r.t to [this documentation](https://docs.google.com/document/d/1XCd5GgbYj-f8psKpfd5kpPwI4D9vpZaedLAiltE3ZWw/edit?pli=1) and [this PR integration](https://github.com/multiversx/mx-chain-sovereign-notifier-go/pull/10). This comp should be able to receive incoming headers and events from a chain to local sovereign chain. This handler will validate the events(using proofs in the future) and create incoming miniblocks and transaction(which will be added in pool) to be executed in sovereign shard.

## Testing procedure
- Local testnet

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: cmd/sovereignnode/testscommon/incomingHeaderStub.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package testscommon
+
+import (
+	"github.com/multiversx/mx-chain-core-go/data"
+	"github.com/multiversx/mx-chain-core-go/data/block"
+)
+
+type IncomingHeaderStub struct {
+	GetHeaderHandlerCalled func() data.HeaderHandler
+}
```
[the remark this answers is not in the document]
Comment (mariusmihaic) [reply cmd/sovereignnode/testscommon/incomingHeaderStub.go:10]: Added it

File: cmd/sovereignnode/incomingHeader/incomingHeaderHandler_test.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+	t.Run("nil tx pool, should return error", func(t *testing.T) {
+		args := createArgs()
+		args.TxPool = nil
+
+		handler, err := NewIncomingHeaderHandler(args)
+		require.Equal(t, errNilTxPool, err)
+		require.Nil(t, handler)
+	})
+
+	t.Run("nil headers pool, should return error", func(t *testing.T) {
```
Comment (raduchis) [inline cmd/sovereignnode/incomingHeader/incomingHeaderHandler_test.go:50]: nil marshaller

File: cmd/sovereignnode/incomingHeader/incomingHeaderHandler_test.go [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+		args := createArgs()
+		args.Marshaller = nil
+
+		handler, err := NewIncomingHeaderHandler(args)
+		require.Equal(t, core.ErrNilMarshalizer, err)
+		require.Nil(t, handler)
+	})
+
+	t.Run("nil headers pool, should return error", func(t *testing.T) {
```
Comment (raduchis) [inline cmd/sovereignnode/incomingHeader/incomingHeaderHandler_test.go:59]: nil hasher

File: cmd/sovereignnode/incomingHeader/incomingHeaderHandler.go [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+	headersPool HeadersPool
+	txPool      TransactionPool
+	marshaller  marshal.Marshalizer
+	hasher      hashing.Hasher
+}
+
+// NewIncomingHeaderHandler creates an incoming header handler which should be able to receive incoming headers and events
+// from a chain to local sovereign chain. This handler will validate the events(using proofs in the future) and create
+// incoming miniblocks and transaction(which will be added in pool) to be executed in sovereign shard.
+func NewIncomingHeaderHandler(args ArgsIncomingHeaderHandler) (*incomingHeaderHandler, error) {
```
[the remark this answers is not in the document]
Comment (mariusmihaic) [reply cmd/sovereignnode/incomingHeader/incomingHeaderHandler.go:37]: It is actually not creating incoming headers. It receives incoming headers and creates `ShardExtendedHeader`.
Renamed it to `IncomingHeaderProcessor`, though

File: cmd/sovereignnode/incomingHeader/incomingHeaderHandler_test.go (hunk 2 above)
Comment (mariusmihaic) [reply cmd/sovereignnode/incomingHeader/incomingHeaderHandler_test.go]: Right, refactored

Comment (raduchis) [reply cmd/sovereignnode/incomingHeader/incomingHeaderHandler_test.go]: ok!

File: cmd/sovereignnode/incomingHeader/incomingHeaderHandler_test.go (hunk 3 above)
Comment (raduchis) [reply cmd/sovereignnode/incomingHeader/incomingHeaderHandler_test.go]: ok!

File: cmd/sovereignnode/incomingHeader/incomingHeaderHandler.go [hunk 5]
```diff
@@ -24,17 +24,17 @@ type ArgsIncomingHeaderHandler struct {
...
-type incomingHeaderHandler struct {
+type incomingHeaderProcessor struct {
 	headersPool HeadersPool
 	txPool      TransactionPool
 	marshaller  marshal.Marshalizer
 	hasher      hashing.Hasher
 }
 
-// NewIncomingHeaderHandler creates an incoming header handler which should be able to receive incoming headers and events
+// NewIncomingHeaderProcessor creates an incoming header handler which should be able to receive incoming headers and events
```
Comment (raduchis) [inline cmd/sovereignnode/incomingHeader/incomingHeaderHandler.go:34]: creates an incoming header **processor**

Comment (mariusmihaic) [reply cmd/sovereignnode/incomingHeader/incomingHeaderHandler.go]: Right, missed it

**Review by raduchis — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Pull request merged**